docs/repo-guides/release-nnf-sw/release-all.md

```diff
@@ -0,0 +1,157 @@
+
+Each component under `nnf-deploy` needs to be released first, then `nnf-deploy` can be updated to
+point to those release versions, then `nnf-deploy` itself can be updated and released.
+
+The documentation repo (NearNodeFlash/NearNodeFlash.github.io) is released separately and is not
+part of `nnf-deploy`, but it should match the version number of `nnf-deploy`. Release this like the
+other components.
+
+- [NearNodeFlash/nnf-deploy](https://github.com/NearNodeFlash/nnf-deploy)
+
+  - [DataWorkflowServices/dws](https://github.com/DataWorkflowServices/dws)
+  - [HewlettPackard/lustre-csi-driver](https://github.com/HewlettPackard/lustre-csi-driver)
+  - [NearNodeFlash/lustre-fs-operator](https://github.com/NearNodeFlash/lustre-fs-operator)
+  - [NearNodeFlash/nnf-mfu](https://github.com/NearNodeFlash/nnf-mfu)
+  - [NearNodeFlash/nnf-ec](https://github.com/NearNodeFlash/nnf-ec)
+  - [NearNodeFlash/nnf-sos](https://github.com/NearNodeFlash/nnf-sos)
+  - [NearNodeFlash/nnf-dm](https://github.com/NearNodeFlash/nnf-dm)
+  - [NearNodeFlash/nnf-integration-test](https://github.com/NearNodeFlash/nnf-integration-test)
+
+- [NearNodeFlash/NearNodeFlash.github.io](https://github.com/NearNodeFlash/NearNodeFlash.github.io)
+
+## Overview of release-all tool
+
+`release-all.sh` automates most of the steps of releasing NNF software and adds additional checks for common issues.
+
+## Assumptions
+
+- `master` or `main` branch for each repository contains **tested** software and documentation ready to be released.
+- You've installed the GitHub CLI tool, `gh`.
+  - This tool requires a GH_TOKEN environment variable containing a `repo` scope classic token.
+
+## Steps
+
+### Run the steps in this order
+
+> **Note:** You almost always want to use the -R option to focus the `phase` activity to a specific repo.
+
+0. **List Repos:** Get the ordered list of repo names to use with -R option in subsequent steps. This is referred to as `repo-list`
+    > **Pro tip:** Keep this list in a separate window for easy viewing
+    ./release-all.sh -L
+
+1. **Check Vendoring:** For each repo's master/main branch; determine whether any of them need to be re-vendored.
+    > **Note:** Ensure each repo is error-free before proceeding to the next repo in `repo-list`
+
+    ```bash
+    For each repo in `repo-list`
+        ./release-all.sh -P master -R $repo
+    ```
+
+2. **Create Trial Release Branch:** Create the new release branch, merge master/main to that release branch, but don't push it yet. The point of this step is to look for merge conflicts between master/main and the release branch.
+
+    ```bash
+    For each repo in `repo-list`
+        ./release-all.sh -P release -R $repo
+    ```
+
+3. **Generate Release:** For each repo in `repo-list`, proceed through the following steps in sequence before moving on to the next repo.
+    > **Note:** The next steps use the gh(1) GitHub CLI tool and require a GH_TOKEN environment variable containing a 'repo' scope classic token.
```

Comment:
what's the (1) after gh referring to? step 1?

Reply:
That's nomenclature for referring to section 1 of the manpages.

Reply:
Notice that the 'gh' manpage is in section 1:
```
$ man gh | sed 1q
GH(1)                          GitHub CLI manual                         GH(1)
```

Reply:
```
$ man man | grep -A10 'The sections of the manual are'
     The sections of the manual are:
           1.   General Commands Manual
           2.   System Calls Manual
           3.   Library Functions Manual
           4.   Kernel Interfaces Manual
           5.   File Formats Manual
           6.   Games Manual
           7.   Miscellaneous Information Manual
           8.   System Manager's Manual
           9.   Kernel Developer's Manual
```

Reply:
For example:
```
$man ls | col -b | grep -A3 'SEE ALSO'
SEE ALSO
     chflags(1), chmod(1), sort(1), xterm(1), localeconv(3), strftime(3),
     strmode(3), compat(5), termcap(5), sticky(7), symlink(7)
```

Reply:
Ah thank you. For some reason that didn't jump out to me in this context.